### PR TITLE
Do not delete subscription type deployables

### PR DESCRIPTION
**Pull request by rokej**

### CODE REVIEW

File: pkg/utils/utilconsts.go [hunk 1]
```diff
@@ -23,8 +23,10 @@ const (
 	debugLevel = klog.Level(5)
 	// HelmCRKind is kind of the Helm CR
 	HelmCRKind = "HelmRelease"
+	// SubscriptionCRKind is kind of the Subscription CR
+	SubscriptionCRKind = "Subscription"
 	// HelmCRAPIVersion is APIVersion of the Helm CR
-	HelmCRAPIVersion = "app.ibm.com/v1alpha1"
+	HelmCRAPIVersion = "apps.open-cluster-management.io/v1"
```
Comment (tengqm) [inline pkg/utils/utilconsts.go]:
```go
import (
  chnv1 ""github.com/open-cluster-management/multicloud-operators-channel/pkg/apis/apps/v1"
)
HelmCRAPIVersioon = chnv1.SchemeGroupVersion.String()
```

Comment (rokej) [reply pkg/utils/utilconsts.go]: I cannot use chnv1.SchemeGroupVersion variable as a constant value.

**Comment by rokej** [comment]
@ianzhang366 
HelmCRAPIVersion = "apps.open-cluster-management.io/v1" change was not part of this PR scope but I noticed it was wrong so changed. There are multiple places in this repo that uses this constant and probably other repos too. We should clean this up in a separate PR.

**Comment by rokej** [comment]
This PR is to support having multiple channels and subscriptions in the same namespace by:

- Not deleting subscription kind deployables from channel namespace

- Adding annotations and labels to synchronizer-generated deployables to identify which channel they belong to

**Comment by ianzhang366** [comment]
FYI, 

I feel we need to use `ownerReference` more often than annotation, especially when handling the ownership kind relationship. I don't know why we didn't use `ownerReference` at the beginning, maybe @xiangjingli  can provide more input.

Using the `ownerReference` we can leverage the k8s GC and help us write complicate deletion logic from parent and child resources.

Also, I feel we might need to revisit the deployable controller at channel repo as well to make sure the logic for multi-channel per namespace is consistent for namespace type channel as well. Maybe ObjectStore type should be revisited as well.

**Comment by tengqm** [comment]
My perception regarding the question related to `ownerReference` is that we are cautious on handling over the lifecycle management to Kubernetes. If we have `ownerReference` set, Kubernetes will step in when determining if something should be garbage collected.
With that said, the flooding of annotations is very confusing. For example, we have "hosting-deployable", "controlled-by", "source-object" and many other annotations. Does this mean an object is supposed to have multiple owners? Annotations are cheap, but we may still want to refrain from adding them freely.
Just my $0.02.

File: pkg/utils/utilconsts.go (hunk 1 above)
Comment (tengqm) [reply pkg/utils/utilconsts.go]: Then make it a var rather than a "const"? :)

**Pull request merged**